toutv/bos.py

```diff
@@ -520,13 +521,24 @@ def _get_playlist_url(self):
         params = dict(toutv.config.TOUTV_PLAYLIST_PARAMS)
         params['idMedia'] = self.PID
 
-        r = self._do_request(url, params=params)
-        response_obj = r.json()
+        num_tries = 0
 
-        if response_obj['errorCode']:
-            raise RuntimeError(response_obj['message'])
+        def finished():
+            return num_tries == 3
 
-        return response_obj['url']
+        while not finished():
+            r = self._do_request(url, params=params)
+            try:
+                response_obj = r.json()
+                if response_obj['errorCode']:
+                    raise RuntimeError(response_obj['message'])
+                return response_obj['url']
+            except ValueError:
+                num_tries += 1
+                if not finished():
+                    print("GetPlaylistURL failed. Will retry...", file=sys.stderr)
+
+        raise RuntimeError("Error: GetPlaylistURL failed.")
```

Comment:
Instead of throwing a new exception, it would be preferrable at least to throw the original exception, it will be easier to track down any problem.

We can always throw our own exception, but then we should use exception chaining:
```py
except ValueError as e:
    raise SomeError(...) from e
```